homeassistant/components/plex/config_flow.py

```diff
@@ -0,0 +1,215 @@
+"""Config flow for Plex."""
+import logging
+
+import plexapi.exceptions
+import requests.exceptions
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import (
+    CONF_URL,
+    CONF_HOST,
+    CONF_PORT,
+    CONF_TOKEN,
+    CONF_SSL,
+    CONF_VERIFY_SSL,
+)
+
+from .const import (
+    CONF_SERVER,
+    CONF_SERVER_IDENTIFIER,
+    DEFAULT_PORT,
+    DEFAULT_SSL,
+    DEFAULT_VERIFY_SSL,
+    DOMAIN,
+    PLEX_SERVER_CONFIG,
+    SERVERS,
+)
+from .errors import NoServersFound, ServerNotSpecified
+from .server import PlexServer
+
+_LOGGER = logging.getLogger(__package__)
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class PlexFlowHandler(config_entries.ConfigFlow):
+    """Handle a Plex config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self):
+        """Initialize the Plex flow."""
+        self.current_login = {}
+        self.discovery_info = {}
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        errors = {}
+
+        if user_input is not None:
+            manual_setup = user_input.get("manual_setup")
+            if manual_setup is True:
+                return await self.async_step_manual_setup()
+
+            self.current_login = user_input
+
+            plex_server = await self.hass.async_add_executor_job(PlexServer, user_input)
+            try:
+                await self.hass.async_add_executor_job(plex_server.connect)
+            except NoServersFound:
+                errors["base"] = "no_servers"
+            except ServerNotSpecified as available_servers:
+                return self.async_show_form(
```

Comment:
If we want to go to a different step we return and await that step. We don't show the form of the step. Each step is responsible for its own form.